### DISCUSSION TITLE
Why did the v-bind="$props" stopped to work on Vue 3?

### DISCUSSION THREAD

**Discussion by K-Schaeffer** [storybookjs/storybook#23798] [Help]
Updating storybook from [7.0.11](https://github.com/storybookjs/storybook/releases/tag/v7.0.11) to [7.1.0](https://github.com/storybookjs/storybook/releases/tag/v7.1.0) I noticed that my stories using **$props** to bind stopped to work, e.g.:
```javascript
const Template = (_, { argTypes }) => ({
  components: { MyComponent },
  props: Object.keys(argTypes),
  template: `
    <MyComponent
      v-bind="$props"
    />
  `,
  },
});
```
![image](https://github.com/storybookjs/storybook/assets/37464395/399147ac-37d7-4fc8-9e19-0424d6a4d3e9)

Basically, all binds stop to work as they used to. With that, I had to refactor to the **args** syntax, e.g.:
```javascript
const Template = (args) => ({
  components: { MyComponent },
  setup() {
    return { args };
  },
  template: `
    <MyComponent
      v-bind="args"
    />
  `,
});
```
![image](https://github.com/storybookjs/storybook/assets/37464395/92c19ab1-6e14-42d5-8304-9e5d2e9b904e)

Doing some research about this I found this PR https://github.com/storybookjs/storybook/pull/13981, which basically added this new syntax to Storybook. What's odd is that this addition was released into 6.x and testing here I could check that until 7.1.0-alpha.10 it worked, so the changes made on 7.1.0-alpha.11 broke this functionality, probably on this commit https://github.com/storybookjs/storybook/commit/8a502243738a9ad4c7a73cf2a58e9de2aed5efd9.

So, to summarize, the purpose of this discussion is:
1. Understand if this breakiing change was really intended, as the releases don't mention that specifically after 6.x version that released the 13981;
2. Check why the actions syntax doesn't work with args binding and if it makes sense, e.g.:
  Used to work:
  ```javascript
  export default {
  component: MyComponent,
  argTypes: {
    myEvent: {
      action: 'my-event',
      control: { disable: true },
    },
  },
};

const Template = (args, { argTypes }) => ({
  components: { MyComponent },
  props: Object.keys(argTypes),
  template: `
    <ReadingQuiz
      v-bind="$props"
      @myEvent="myEvent"
    />
  `,
});
  ```
  Current way of making it work:
  ```javascript
  export default {
  component: MyComponent,
  argTypes: {
  },
};

const Template = (args) => ({
  components: { MyComponent },
  setup() {
    return { args };
  },
  methods: {
    myEvent(payload) {
      action('my-event')(payload);
    },
  },
  template: `
    <ReadingQuiz
      v-bind="args"
      @myEvent="myEvent"
    />
  `,
});
  ```
This issue was also mentioned here https://github.com/storybookjs/storybook/issues/18688.

So that's pretty much it, the few disadvantages I currently see on using the args syntax are the requirement of using this legacy action syntax with methods and I also used to face some reactivity issues on some edge cases, though I couldn't really reproduce them so we might count only this actions thing as a current disadvantage.

**Comment by Dtsiantaris** [comment]
Have you actually figured this out? I cannot make both slots and props work together.
```
const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BSelect },
  setup() {
    return { args };
  },
  template: `
    <BSelect v-bind="args">
    {{ args.default }}
    </BSelect>
    `,
});
```

This is the only decent working template. But trying to add `options`, which is an array, like that:
```
export const Secondary = Template.bind({});
Secondary.args = {
  label: "Secondary label",
  options: [
    { value: "foo", label: "fooLabel" },
    { value: "bar", label: "barLabel" },
  ],
};
```

produces this code:
![image](https://github.com/storybookjs/storybook/assets/44876922/781965e4-9c0b-42bb-87f2-11dc2ae804c4)

Which makes the options not rendering.

**Comment by joaopedrodcf** [comment]
This helped me a lot thank you @K-Schaeffer  🎖️
